server/src/test/java/org/apache/druid/discovery/BaseNodeRoleWatcherTest.java

```diff
@@ -141,17 +146,18 @@ public void testRegisterListenerBeforeTimeout() throws InterruptedException
     nodeRoleWatcher.childRemoved(broker2);
 
     assertListener(listener1, false, Collections.emptyList(), Collections.emptyList());
+    Assert.assertFalse(listener1.nodeViewInitializationTimedOut.get());
```

Comment:
I think this can be removed.

Reply:
In general, this assert too relies on timing of events. something we should avoid if we can.